### PR TITLE
DATAUP-389 - refactor RunJob unit tests

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

Factors out some mocking that will be in common with other tests once
the job requirements resolution code will be integrated. Should make the
upcoming changes easier to understand.

# Jira Ticket / Github Issue #
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [x] Tests pass in Github Actions and locally 
- [x] Changes available by spinning up a local test suite

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - 3k warnings
- [n/a] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [x] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the Github Actions build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: test/tests_for_sdkmr/EE2Runjob_test.py [hunk 1]
```diff
@@ -4,183 +4,223 @@
...
-        "method": method,
-        "source_ws_objects": [ws_obj1, ws_obj2],
-    }
-    assert rj.run(params, as_admin=True) == job_id
 
-    # check mocks called as expected. The order here is the order that they're called in the code.
-    sdkmr.check_as_admin.assert_called_once_with(JobPermissions.WRITE)
-    ws.get_object_info3.assert_called_once_with(
-        {"objects": [{"ref": ws_obj1}, {"ref": ws_obj2}], "ignoreErrors": 1}
+def _check_common_mock_calls(mocks, reqs, creqs, wsid):
```
Comment (MrCreosote) [inline test/tests_for_sdkmr/EE2Runjob_test.py:95]: creqs will be removed once the requirements resolver code is integrated

File: test/tests_for_sdkmr/EE2Runjob_test.py [hunk 2]
```diff
@@ -4,183 +4,223 @@
...
         {"module_name": "lolcats", "version": "release"}
     )
 
     # initial job data save
     expected_job = Job()
-    expected_job.user = user
-    expected_job.status = created_state
+    expected_job.user = _USER
+    expected_job.status = _CREATED_STATE
+    expected_job.wsid = wsid
```
Comment (ialarmedalien) [inline test/tests_for_sdkmr/EE2Runjob_test.py:113]: worth making into a variable? it's used a couple of lines further down.

**Review by ialarmedalien — APPROVED** [review]
lgtm

File: test/tests_for_sdkmr/EE2Runjob_test.py (hunk 2 above)
Comment (MrCreosote) [reply test/tests_for_sdkmr/EE2Runjob_test.py:113]: Is this comment on the wrong line maybe? `wsid` is a variable...

Comment (MrCreosote) [reply test/tests_for_sdkmr/EE2Runjob_test.py:113]: Resolved this offline, leaving as is for now, anyway

**Pull request merged**